osu.Game.Rulesets.Osu/UI/Cursor/OsuCursorContainer.cs

```diff
@@ -37,9 +43,33 @@ public OsuCursorContainer()
         }
 
         [BackgroundDependencyLoader(true)]
-        private void load(OsuRulesetConfigManager config)
+        private void load(OsuConfigManager config, OsuRulesetConfigManager rulesetConfig, IBindable<WorkingBeatmap> beatmap)
+        {
+            rulesetConfig?.BindWith(OsuRulesetSetting.ShowCursorTrail, showTrail);
+
+            this.beatmap.BindTo(beatmap);
+            this.beatmap.ValueChanged += _ => calculateScale();
+
+            cursorScale = config.GetBindable<float>(OsuSetting.GameplayCursorSize);
+            cursorScale.ValueChanged += _ => calculateScale();
+
+            autoCursorScale = config.GetBindable<bool>(OsuSetting.AutoCursorSize);
+            autoCursorScale.ValueChanged += _ => calculateScale();
+
+            calculateScale();
+        }
+
+        private void calculateScale()
         {
-            config?.BindWith(OsuRulesetSetting.ShowCursorTrail, showTrail);
+            float scale = cursorScale.Value;
+
+            if (autoCursorScale.Value && beatmap.Value != null)
+            {
+                // if we have a beatmap available, let's get its circle size to figure out an automatic cursor scale modifier.
+                scale *= 1f - 0.7f * (1f + beatmap.Value.BeatmapInfo.BaseDifficulty.CircleSize - BeatmapDifficulty.DEFAULT_DIFFICULTY) / BeatmapDifficulty.DEFAULT_DIFFICULTY;
+            }
+
+            ActiveCursor.Scale = cursorTrail.Scale = new Vector2(scale);
```

Comment:
What about using this:
```suggestion
            Scale = new Vector2(scale);
            Size = 1 / Scale;
```

Reply:
Drawables generally shouldn't be changing their own properties. If a consumer were to set scale/size on this container, the changes you proposed would reset them.

I prefer the original change.